File: frontend/src/api/trainingApi.js
import API from "./api";

export async function createSensitivityRecommendation(payload) {
  const { data } = await API.post("/training/sensitivity/recommend", payload);
  return data;
}

export async function fetchSensitivityHistory(limit = 20) {
  const { data } = await API.get("/training/sensitivity/history", { params: { limit } });
  return data;
}

export async function createInputAnalysis(payload) {
  const { data } = await API.post("/training/input/analyze", payload);
  return data;
}

export async function fetchInputHistory(limit = 20) {
  const { data } = await API.get("/training/input/history", { params: { limit } });
  return data;
}

File: backend/app/training_api.py
"""Local-only training APIs."""

from __future__ import annotations

from fastapi import APIRouter, Query

from .env_utils import resolve_config_path
from .sensitivity_lab import (
    SensitivityRecommendationRequest,
    recommend_sensitivity,
)
from .training_db import TrainingDB


router = APIRouter(prefix="/api/training", tags=["training"])
_training_db: TrainingDB | None = None


def get_training_db() -> TrainingDB:
    global _training_db
    if _training_db is None:
        _training_db = TrainingDB(resolve_config_path().parent / "cs2-insight.db")
    return _training_db


@router.post("/sensitivity/recommend")
async def create_sensitivity_recommendation(body: SensitivityRecommendationRequest):
    recommendation = recommend_sensitivity(body)
    return await get_training_db().save_sensitivity_session(
        body.model_dump(mode="json"),
        recommendation.model_dump(mode="json"),
    )


@router.get("/sensitivity/history")
async def list_sensitivity_history(limit: int = Query(default=20, ge=1, le=100)):
    return {"items": await get_training_db().list_sensitivity_sessions(limit)}
